service/history/execution/mutable_state_task_generator.go

```diff
@@ -554,3 +568,18 @@ func (r *mutableStateTaskGeneratorImpl) getTargetDomainID(
 
 	return targetDomainID, nil
 }
+
+func getNextDecisionTimeout(attempt int64, defaultStartToCloseTimeout time.Duration) time.Duration {
+	if attempt <= 1 {
+		return defaultStartToCloseTimeout
+	}
+
+	nextInterval := float64(defaultInitIntervalForDecisionRetry) * math.Pow(2, float64(attempt-2))
```

Comment:
Don't we have an exponential backoff library we can use here?

Reply:
backoff.Retrier I think

Reply:
backoff.Retrier is kind of stateful with attempt recorded. 
Here we need a "stateless" backoff just to getNextInterval with given attempt. (we cannot attach one retrier to taskGenerator to achieve this because retrier doesn't know which task it is retrying)